science/plumed/files/0001-Added-enable-libsearch-to-configure.patch

```diff
@@ -0,0 +1,420 @@
++# listed in LDFLAGS are linked
++PLUMED_CONFIG_ENABLE([libsearch],[libsearch],[search for libraries],[yes])
++
+ PLUMED_CONFIG_ENABLE([doc],[doc],[documentation],[yes])
+ PLUMED_CONFIG_ENABLE([pdfdoc],[pdfdoc],[pdf version of the manual],[no])
+ PLUMED_CONFIG_ENABLE([debug],[debug],[debugging],[no])
+@@ -344,9 +354,12 @@ fi
+ if test "$external_blas" == true ; then
+ AC_CHECK_FUNC(  [dgemv], [blas_found=nounderscore], [
+ AC_CHECK_FUNC(  [dgemv_],[blas_found=underscore],   [
+-AC_CHECK_LIB(   [blas],[dgemv],  [LIBS="-lblas $LIBS"] [blas_found=nounderscore], [
+-AC_CHECK_LIB(   [blas],[dgemv_], [LIBS="-lblas $LIBS"] [blas_found=underscore]
+-) ]) ]) ])
++if test "${libsearch}" == true ; then
++  AC_CHECK_LIB(   [blas],[dgemv],  [LIBS="-lblas $LIBS"] [blas_found=nounderscore], [
++  AC_CHECK_LIB(   [blas],[dgemv_], [LIBS="-lblas $LIBS"] [blas_found=underscore]
++) ])
++fi
++]) ])
+ fi
+ 
+ # if not found, then use internal lapack and blas
+@@ -370,8 +383,9 @@ case "$blas_found" in
+ (nounderscore) search_for=dsyevr ;;
+ esac
+ AC_CHECK_FUNC( [$search_for], [lapack_found=$blas_found], [
+-AC_CHECK_LIB(  [lapack],[$search_for], [LIBS="-llapack $LIBS"] [lapack_found=yes]
+-)
++if test "${libsearch}" == true ; then
++  AC_CHECK_LIB(  [lapack],[$search_for], [LIBS="-llapack $LIBS"] [lapack_found=yes])
++fi
+ ])
+ fi
+ 
+@@ -418,7 +432,10 @@ AC_DEFINE([__PLUMED_HAS_MOLFILE_PLUGINS])
+ found=ko
+ if test "$external_molfile_plugins" == true ; then
+ 	AC_CHECK_LIB([molfile_plugin],[molfile_dcdplugin_init],
+-        	[ AC_CHECK_HEADER([libmolfile_plugin.h], [LIBS="-lmolfile_plugin $LIBS"] [found=ok] ) ])
++           if test "${libsearch}" == true ; then
++        	[ AC_CHECK_HEADER([libmolfile_plugin.h], [LIBS="-lmolfile_plugin $LIBS"] [found=ok] ) ]
++           fi
++)
+ fi
+ if test $found == ko ; then
+ 	AC_MSG_WARN([using internal molfile_plugins, which only support dcd/xtc/trr/trj/crd files])
+@@ -480,14 +497,18 @@ save_LIBS="$LIBS"
+ if test $gsl == true ; then
+   found=ko
+   AC_CHECK_FUNC(  [cblas_dgemv], [found=ok], [
+-  AC_CHECK_LIB(   [gslcblas],[cblas_dgemv], [LIBS="-lgslcblas $LIBS"] [found=ok]
+-  ) ])
++  if test "${libsearch}" == true ; then
++    AC_CHECK_LIB(   [gslcblas],[cblas_dgemv], [LIBS="-lgslcblas $LIBS"] [found=ok])
++  fi
++  ])
+   if test $found == ok ; then
+     found=ko
+     AC_CHECK_HEADER(  [gsl/gsl_vector.h], [
+       AC_CHECK_FUNC(  [gsl_vector_alloc], [found=ok], [
+-      AC_CHECK_LIB(   [gsl],[gsl_vector_alloc], [LIBS="-lgsl $LIBS"] [found=ok]
+-      ) ])
++      if test "${libsearch}" == true ; then
++        AC_CHECK_LIB(   [gsl],[gsl_vector_alloc], [LIBS="-lgsl $LIBS"] [found=ok])
++      fi
++    ])
+     ])
+   fi
+   if test $found == ok ; then
+@@ -760,7 +781,12 @@ EOF
+     $CXX $CXXFLAGS -c conftest.cpp 
+ # start search:
+     found=
+-    for testlib in "" -lstdc++ -lc++ -lmpi_cxx ; do
++    if test "${libsearch}" == true ; then
++      testlibs="-lstdc++ -lc++ -lmpi_cxx"
```

Comment:
This set of libraries is very weird (but I admit that I'm a bit ignorant in this area). Are you simultaneously including `libc++` and `libstdc++`?

If that is the case, you should respect the variable that's given to you with `${configure.cxx_stdlib}`.